src/Libraries/LibAllowList.sol

```diff
@@ -6,147 +6,308 @@ import { LibAsset } from "./LibAsset.sol";
 
 /// @title LibAllowList
 /// @author LI.FI (https://li.fi)
-/// @notice Library for managing and accessing the contract address allow list
+/// @notice Manages a dual-model allow list to support a secure, granular permissions system
+/// while maintaining backward compatibility with a non-granular, global system.
+/// @dev This library is the single source of truth for all whitelist state changes.
+/// It ensures that both the new granular mapping and the global arrays used by older
+/// contracts are kept perfectly synchronized. The long-term goal is to migrate all usage
+/// to the new granular system. New development should exclusively use the "Primary Interface" functions.
+///
+/// Special Marker Selector:
+/// - Contracts with no specific function selectors use 0xffffffff as a marker selector.
```

Comment:
also here, please lets be more clear. 
contracts that are whitelisted as approveTo only (but no functions can be called on these contracts) 

(sth like this)